plugins/modules/aci_dhcp_relay.py

```diff
@@ -0,0 +1,275 @@
+                       "ownerKey": "",
+                       "ownerTag": ""
+                   }
+               }
+           }
+       ]
+   proposed:
+     description: The assembled configuration from the user-provided parameters
+     returned: info
+     type: dict
+     sample:
+       {
+           "fvTenant": {
+               "attributes": {
+                   "descr": "Production environment",
+                   "name": "production"
+               }
+           }
+       }
+   filter_string:
+     description: The filter string used for the request
+     returned: failure or debug
+     type: str
+     sample: ?rsp-prop-include=config-only
+   method:
+     description: The HTTP method used for the request to the APIC
+     returned: failure or debug
+     type: str
+     sample: POST
+   response:
+     description: The HTTP response from the APIC
+     returned: failure or debug
+     type: str
+     sample: OK (30 bytes)
+   status:
+     description: The HTTP status from the APIC
+     returned: failure or debug
+     type: int
+     sample: 200
+   url:
+     description: The HTTP url used for the request to the APIC
+     returned: failure or debug
+     type: str
+     sample: https://10.11.12.13/api/mo/uni/tn-production.json
+   '''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        name=dict(type='str', aliases=['relay_policy']),
+        description=dict(type='str'),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+        tenant=dict(type='str'),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['name', 'tenant']],
+            ['state', 'present', ['name', 'tenant']],
+        ],
+    )
+
+    name = module.params.get('name')
+    description = module.params.get('description')
+    state = module.params.get('state')
+    tenant = module.params.get('tenant')
+    child_classes = ['dhcpRsProv']
+
+    aci = ACIModule(module)
+    aci.construct_url(
+        root_class=dict(
+            aci_class='fvTenant',
+            aci_rn='tn-{0}'.format(tenant),
+            module_object=tenant,
+            target_filter={'name': tenant},
+        ),
+        subclass_1=dict(
+            aci_class='dhcpRelayP',
+            aci_rn='relayp-{0}'.format(name),
+            module_object=name,
+            target_filter={'name': name},
+        ),
+        child_classes=child_classes,
+    )
+
+    aci.get_existing()
+
+    if state == 'present':
+
+        aci.payload(
+            aci_class='dhcpRelayP',
+            class_config=dict(
+                name=name,
+                descr=description,
+                owner='tenant',
+                mode='visible',
```

Comment:
mode visible is the default so it is not needed. I would remove it from here.

Reply:
removed.